lib/player_turn.rb

```diff
@@ -0,0 +1,36 @@
+require_relative "coordinates"
+require_relative "null_coordinates"
+
+class PlayerTurn
+  attr_reader :input, :view
+
+  def initialize(input:, view:)
+    @input = input
+    @view = view
+  end
+
+  def get_coordinates
+    until coordinates.valid?
```

Comment:
This is so sweet! I love how the null object cleared up so much logic.

Reply:
Yeah, that part was enjoyable.